### ISSUE TITLE
Proyecto: TrueRisk - Inteligencia climática para salvar vidas en España

### ISSUE THREAD

**Issue by javierdejesusda** [midudev/hackaton-cubepath-2026#279]
**📛 Nombre del proyecto:**

TrueRisk - Inteligencia climática para salvar vidas en España

---

**📝 Descripción del proyecto:**

El 29 de octubre de 2024, una DANA arrasó Valencia. 229 personas murieron en una sola noche. Pueblos enteros quedaron bajo el agua. Las alertas llegaron tarde o nunca llegaron.

Ese día me hice una pregunta: ¿y si la tecnología pudiera haber avisado a tiempo?

TrueRisk nació de esa pregunta. No es un proyecto solo de hackathon, es la respuesta técnica a un problema que mató a 229 personas y que volverá a ocurrir. España es el país europeo más expuesto a desastres climáticos, y no existe ninguna plataforma pública que integre predicción, alerta y preparación en un solo lugar.

## Qué hace TrueRisk

Plataforma de inteligencia de riesgo climático que monitoriza las 52 provincias de España en tiempo real, cubriendo 8 tipos de amenazas naturales: inundaciones, incendios forestales, sequías, olas de calor, eventos sísmicos, olas de frío, tormentas de viento y eventos compuestos tipo DANA. TrueRisk no busca reemplazar a las autoridades, sino darles a los ciudadanos los segundos que marcan la diferencia.

### El cerebro: 7 modelos de Machine Learning en paralelo

Cada amenaza tiene su propio modelo especializado entrenado con datos reales:

| Amenaza | Modelo | Features | Accuracy | F1 | AUC-ROC |
|---------|--------|----------|----------|-----|---------|
| Inundación | XGBoost | 23 | 89% | 0.84 | 0.93 |
| Incendio | RF + LightGBM | 20 | 91% | 0.87 | 0.95 |
| Sequía | SPEI + LSTM (PyTorch) | 6 (90 días) | 86% | 0.81 | 0.90 |
| Ola de calor | XGBoost + WBGT | 18 | 88% | 0.83 | 0.92 |
| Sísmico | Rule-based (IGN) | 8 | 92% | 0.78 | — |
| Ola de frío | TFT (Temporal Fusion Transformer) | 14 | 90% | 0.76 | — |
| Tormenta | Rule-based + patrones sinópticos | 14 | 91% | 0.79 | — |

No usamos un único modelo genérico, cada amenaza tiene su pipeline de feature engineering con +26 features temporales (acumulación de precipitación a 1h/6h/24h/48h/7d, días consecutivos secos/cálidos/fríos, dinámica de presión atmosférica, Canadian FWI, SPEI, índices de estrés térmico WBGT/UTCI). 

### Arquitectura avanzada: TFT + Graph Neural Networks

- **Temporal Fusion Transformers (TFT)** con Variable Selection Networks, Gated Residual Networks y atención multi-cabeza interpretable para forecasting multi-horizonte (6h, 12h, 24h, 48h, 72h, 168h)
- **Graph Attention Networks (GATv2)** sobre un grafo de 52 provincias con aristas basadas en redes fluviales, propagando riesgo entre regiones hidrológicamente conectadas
- **Motor de riesgo compuesto** inspirado en la metodología INFORM de Naciones Unidas

### Video 

Cuenta la historia completa, desde la tragedia de la DANA hasta cada funcionalidad:

https://github.com/user-attachments/assets/9cad435c-6cd7-40da-af6d-ed9437fdba40

Publicamos un paper técnico documentando la arquitectura: [Technical Report (PDF)](https://github.com/javierdejesusda/TrueRisk/blob/main/docs/technical-report/main.pdf)

### Funcionalidades para el ciudadano

- 🗺️ **Mapa interactivo** con niveles de riesgo por provincia, alertas activas y actividad sísmica en tiempo real
- 📊 **Dashboard personalizado** con scores 0-100, desglose por amenaza y explicabilidad del modelo
- 🤖 **Asesor de emergencias con IA** con orientación contextual basada en condiciones actuales
- 🚨 **Alertas multicanal**: Web Push, SMS, Telegram bot, email
- 🏠 **Evaluación de riesgo a nivel de propiedad** con informes PDF para propietarios y aseguradoras
- 🚗 **Rutas de evacuación** adaptadas a condiciones de riesgo actuales
- 👥 **Comunidad ciudadana** de reporte de peligros
- 🌍 **Bilingüe** español/inglés
- 📈 **Predicciones de sequía** con proyecciones climáticas
- 🎮 **Gamificación** para incentivar la preparación ante emergencias

### Stack técnico

**Frontend:** Next.js 16 · React 19 · TypeScript · Tailwind CSS v4 · Framer Motion · Zustand · Recharts · MapLibre GL · next-intl · NextAuth v5

**Backend:** Python 3.12 · FastAPI · SQLAlchemy async · Alembic · XGBoost · LightGBM · PyTorch (LSTM + TFT) · torch-geometric (GATv2)

**Datos en tiempo real:** AEMET (alertas CAP) · Open-Meteo (pronósticos) · IGN (catálogo sísmico) · NASA FIRMS (incendios) · OpenAQ (calidad del aire) · Copernicus (clima)

**Infra:** Docker Compose · PostgreSQL 16 · GitHub Actions CI/CD · Dokploy en CubePath · Sentry

**Testing:** Vitest + Playwright · pytest · mypy · ruff · ESLint

52 provincias · 8 modelos ML · 15+ fuentes de datos · Tiempo real · Vidas que proteger.

---

**🔗 URL de la web (desplegada en CubePath):**

https://truerisk.cloud

---

**📦 URL del repositorio (público):**

https://github.com/javierdejesusda/TrueRisk

---

**📸 Capturas de pantalla o GIFs:**

<img width="1914" height="911" alt="Image" src="https://github.com/user-attachments/assets/521a3d2c-f492-4aa8-b190-832fd8f46884" />

<img width="1917" height="942" alt="Image" src="https://github.com/user-attachments/assets/b7bc7fdd-154c-43fb-a143-3130f8ab4389" />

<img width="1917" height="933" alt="Image" src="https://github.com/user-attachments/assets/e1eedc2f-b324-432d-94a7-28efa237e863" />

<img width="595" height="767" alt="Image" src="https://github.com/user-attachments/assets/97862757-a144-45c1-af11-1bb7d9f6eda2" />

Publicación del paper técnico documentando la arquitectura: [Technical Report (PDF)](https://github.com/javierdejesusda/TrueRisk/blob/main/docs/technical-report/main.pdf)

<img width="1088" height="1539" alt="Image" src="https://github.com/user-attachments/assets/58adccfd-8bef-416d-a1ba-c6bcbaba8856" />

---

**☁️ ¿Cómo has utilizado CubePath?:**

TrueRisk corre en producción en un VPS de CubePath con Dokploy como orquestador.

Desplegamos un Docker Compose con 3 servicios: frontend (Next.js 16), backend (FastAPI + 7 modelos ML en PyTorch/XGBoost/LightGBM) y PostgreSQL 16, cada uno con health checks y límites de recursos.

El CI/CD está automatizado end-to-end: cada push a main ejecuta tests en GitHub Actions y, si pasan, dispara el deploy vía la API de Dokploy (`POST /api/compose.deploy`) con verificación de salud post-deploy.

Las 30+ variables de entorno (claves de AEMET, Twilio, Sentry, VAPID, etc.) se gestionan directamente desde Dokploy. El dominio truerisk.cloud apunta al VPS con HTTPS.

---

**📧 Email de contacto:**

javier.dejesusj9@gmail.com - alerts@truerisk.cloud

---

**✅ Confirmación:**
- [x] Mi proyecto está desplegado en CubePath y funciona correctamente
- [x] El repositorio es público y contiene un README con la documentación
- [x] He leído y acepto las reglas de la hackatón

**Comment by javierdejesusda** [comment]
Debido a un problema con el renderizado de las imágenes en la descripción principal, adjunto aquí la documentación visual completa del proyecto:

### Video 

https://www.youtube.com/watch?v=i0kSuHTb1_8

### Capturas

<img width="1914" height="911" alt="Image" src="https://github.com/user-attachments/assets/a0820b24-b7df-47a9-914e-c862b3ff6750" />

<img width="1917" height="942" alt="Image" src="https://github.com/user-attachments/assets/2de7da59-8f1b-4d48-a878-d1343f33d1fb" />

<img width="1917" height="933" alt="Image" src="https://github.com/user-attachments/assets/cf38ffe8-0ca5-4f91-820e-1ada1d7df309" />

<img width="595" height="767" alt="Image" src="https://github.com/user-attachments/assets/0cb6c880-feaa-4a0e-987d-27501a121d84" />

<img width="1088" height="1539" alt="Image" src="https://github.com/user-attachments/assets/8170184a-8669-4980-aec1-1f44624d929a" />
